### ISSUE TITLE
Well... did i kill it? NSPanel died after update and change the bluetooth settings

### ISSUE THREAD

**Issue by TeDeVPrime** [Blackymas/NSPanel_HA_Blueprint#2069]
### TFT Version

4.3.2

### ESPHome Version

4.3.2

### Blueprint Version

4.3.2

### Panel Model

EU

### What is the bug?

i spent a few hours trying to update my nspanel to the latest 4.3.2.
long story short... i had to download the tft locally so that i could actually work.

right when i said "finally!" i decided to try and enable the ble_tracker...

i included these lines

esp32_ble_tracker:
  scan_parameters:
    interval: 1100ms
    window: 1100ms

then i pressed installed in the ESPHome.... 
the installation finished.... and then it died.... it doesn't boot up.... i tried turning the power off and on again ... nothing. it's a brick device.

i know that it doesn't make sense... but this is my story. any ideas?

**Comment by edwardtfn** [comment]
Flash it using TTL adapter and without the Bluetooth and it should work.

By the way, it will probably work with Bluetooth if you use v4.3.4. 😉

**Comment by edwardtfn** [comment]
The idea is to have a BLE Tracker add-on in experimental mode with v4.3.5.

**Comment by TeDeVPrime** [comment]
@edwardtfn hi. what do you mean it will work with bluetooth? i had the latest version installed 4.3.4.
then i tried the above config and it died on me. nothing works.

**Comment by edwardtfn** [comment]
It should work with v4.3.4. Do you have any other component added in your yaml?

**Comment by TeDeVPrime** [comment]
> It should work with v4.3.4. Do you have any other component added in your yaml?

no i don't. this is the only one i installed. so you believe that it is not bricked? i should remove it from the wall and try to connect it with the usb adapter again?

**Comment by edwardtfn** [comment]
Yes, I got to this situation a few times when playing with different components and it's just about compile the firmware with just the basics, then install using TTL and the banel is back to life.

**Comment by edwardtfn** [comment]
The issue is about getting out of RAM during the boot, then the panel stops it even before turning on the display or starting support to OTA flashing.
Flashing with a firmware which didn't take all the memory should work, but will require TTL adapter as OTA is out.

**Comment by TeDeVPrime** [comment]
> The issue is about getting out of RAM during the boot, then the panel stops it even before turning on the display or starting support to OTA flashing. Flashing with a firmware which didn't take all the memory should work, but will require TTL adapter as OTA is out.

ok i will try it next weekend because i don't have time for that right now. i also ordered a new nspanel just to be sure. they came out as an offer yesterday for 30 euros!

**Comment by edwardtfn** [comment]
Where did you found this price?

**Comment by Mauricewillemstein** [comment]
> ok i will try it next weekend because i don't have time for that right now. i also ordered a new nspanel just to be sure. they came out as an offer yesterday for 30 euros!

I also want to know wehere you found it for 30 euros :)

**Comment by TeDeVPrime** [comment]
it's on offer here in Greece for this week. i ordered two
https://www.e-shop.gr/sonoff-nspanel-smart-scene-wall-switch-black-p-PER.941903

**Comment by edwardtfn** [comment]
How is the panel where you got the black screen? Flashing via ttl recovered that one?

**Comment by TeDeVPrime** [comment]
> How is the panel where you got the black screen? Flashing via ttl recovered that one?

ok i just managed to get it back into working again.
how should i enable the bluetooth?

i have the latest TFT and version installed

**Comment by edwardtfn** [comment]
I'd like to know if the add on Bluetooth proxy works for you. That worked fine on all my tests, so I really wanna know if that is working or not.
I will fully understand if you don't wanna give a try, as the chance of having to flash via TTL exists, but if you wanna try that, please let us know your results.

**Comment by TeDeVPrime** [comment]
> I'd like to know if the add on Bluetooth proxy works for you. That worked fine on all my tests, so I really wanna know if that is working or not. I will fully understand if you don't wanna give a try, as the chance of having to flash via TTL exists, but if you wanna try that, please let us know your results.

well i have closed everything again so if i need to flash via ttl again i will cry, since taking the panel off is a real pain the butt.
i will probably try it on another nspanel that i have. 
what configuration do i need to add in the esp in order for the bluetooth to work?

**Comment by Ierlandfan** [comment]
Maybe dumb side question: I I flashed Pre-built firmware update: 8833784908 which addds th BT functionality. Just be sure I have flashed it via TTL (Successfully) I also see a BT icon in the display (gray, just like wireless) 
But how to check whether it is working? Shouldn't I get a "new device"inside Home assistant?

**Comment by edwardtfn** [comment]
> what are the best settings for enable BLE on the nspanel?

Use one of the add-ons available:
- https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/addon_ble_tracker.md
- https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/addon_bluetooth_proxy.md

**Comment by edwardtfn** [comment]
Make sure you are not enforcing arduino (it requires esp-idf).

**Comment by TeDeVPrime** [comment]
can i do it wirelessly or do i need to flash with TTL again?

**Comment by edwardtfn** [comment]
Wirelessly should be fine.

**Comment by TeDeVPrime** [comment]
can i get some help with this?
i managed to install ble_tracker (i am still not sure about the differences between the tracker and the proxy.)

in the long run first i would like to be able to track my smart watch in the house.
secondly i would like to add bluetooth devices into home assistant like the xiaomi mi scale that i have for example.

after installing the ble_tracker, there is nothing that i can see in my integrations. am i doing something wrong here?

**Comment by edwardtfn** [comment]
> in the long run first i would like to be able to track my smart watch in the house.

This is [BLE Tracker](https://esphome.io/components/esp32_ble_tracker.html). You have to setup your watch as a [BLE Client Sensor](https://esphome.io/components/sensor/ble_client) depending on what you wanna do with that.

> secondly i would like to add bluetooth devices into home assistant like the xiaomi mi scale that i have for example.

This is where BLE proxy helps.

**Comment by edwardtfn** [comment]
If your scale is supported by Home Assistant, use that very same integration as if the scale was close to home assistant host. What proxy is doing is extending the Bluetooth coverage using your panel's Bluetooth antena, but it is quite transparent, as if your scale was close to Home Assistant server.

You can also set the integration directly as a customization on ESPHome using the native component: https://esphome.io/components/sensor/xiaomi_miscale.html (this is probably the way I woukd go).
This requires Bluetooth tracker, not proxy.

**Comment by TeDeVPrime** [comment]
so in the esp configuration of the nspanel i need to add the following...

sensor:
  - platform: xiaomi_miscale
    mac_address: '5C:CA:D3:70:D4:A2'
    weight:
      name: "Xiaomi Mi Scale Weight"
    impedance:
      name: "Xiaomi Mi Scale Impedance"
      
but how do i get the mac_address from the scale?

**Comment by TeDeVPrime** [comment]
i installed the bluetooth proxy and it discovered the devices.
now i am just trying to figure out how to track my watch with it

**Comment by Ierlandfan** [comment]
Was working fine and at some point I had to disconnect my Wifi AP. Now it is stuck on BT on and Wifi Off. If I reset the device I do see it's getting a correct IP but it looks like Wifi gets disabled when it fully boots. Is there an option to undo that?

**Comment by plasterer41** [comment]
Worst case scenario you could reflash with cable again. Sent from my iPhoneOn 23 Jun 2024, at 15:35, Ierlandfan ***@***.***> wrote:﻿
Was working fine and at some point I had to disconnect my Wifi AP. Now it is stuck on BT on and Wifi Off. If I reset the device I do see it's getting a correct IP but it looks like Wifi gets disabled when it fully boots. Is there an option to undo that?

**Comment by klibro** [comment]
I have a weird problem with the Bluetooth proxy add on. On one of my nspanels, I added the proxy add on to the yaml and updated ota. It is working perfectly. On another panel, I used the exact same yaml (only difference is the device name), did an ota and it bricked the device. I tried flashing it using a cable, but that didn't work. I only managed to recover the panel by removing the Bluetooth proxy add on line in the yaml, rebuilding and reflashing with the cable. All works again. Both nspanels are running the same version 4.3.9 and firmware 2024.7.3. I noticed in the docs that the partition size needs to be resized and hence ota won't work, but in my case the only panel that works used ota. Is there something special I need to do to size the partition, or is that done automatically when I flash with a cable?

**Comment by plasterer41** [comment]
Reflash by cable with Bluetooth proxy omitted from your yaml. Then upgrade wirelessly if you wish to the version on dev with Bluetooth proxy enabled. Worked perfectly for me when 4.3.9 failed as you have described.

**Comment by klibro** [comment]
Thanks for your response. I'm a bit confused though. I reflashed to basic (no ble proxy), got that working, then flashed (using a cable) to an image with the ble proxy, and that bricked it again. So I'm wondering, why do you think if I do that ota rather than using a cable I'll get a different result? Is there something that ota does that cable-based flashing doesn't? I'm slightly loathe to go through another experiment without understanding why it might work.

**Comment by edwardtfn** [comment]
It's about using `dev`. It should work with both serial or OTA (if you have access to).

**Comment by edwardtfn** [comment]
I will try to find some time to release it ASAP.

**Comment by klibro** [comment]
Thank you so much! I can confirm that ota flashing with the dev branch was successful.

**Comment by plasterer41** [comment]
👍🏼Sent from my iPhoneOn 7 Aug 2024, at 11:53, klibro ***@***.***> wrote:﻿
Thank you so much! I can confirm that ota flashing with the dev branch was successful.

**Comment by wojciechczyz** [comment]
@TeDeVPrime should this issue be closed?
